internal/resources/resource_host.go

```diff
@@ -684,10 +764,47 @@ func resourceMetalHostUpdate(d *schema.ResourceData, meta interface{}) (err erro
 		return err
 	}
 
+	isAsync, err := isHostActionAsync(d)
```

Comment:
This block of new code looks like can be put into a helper function, taking in Pending/Target as input parameter. Then it can be reused for Host create, update, and delete.